data/mods/pokebilities/abilities.js

```diff
@@ -2,38 +2,88 @@
 
 /**@type {{[k: string]: ModdedAbilityData}} */
 exports.BattleAbilities = {
+	powerofalchemy: {
+		inherit: true,
+		onAllyFaint(ally) {
+			let pokemon = this.effectData.target;
+			if (!pokemon.hp) return;
+			let isAbility = pokemon.ability === 'powerofalchemy';
+			/**@type {string[]} */
+			let possibleAbilities = [ally.ability];
+			if (ally.innates) possibleAbilities = possibleAbilities.concat(ally.innates);
+			let bannedAbilities = ['battlebond', 'comatose', 'disguise', 'flowergift', 'forecast', 'illusion', 'imposter', 'multitype', 'powerconstruct', 'powerofalchemy', 'receiver', 'rkssystem', 'schooling', 'shieldsdown', 'stancechange', 'trace', 'wonderguard', 'zenmode'];
+			bannedAbilities.push(pokemon.ability);
+			if (pokemon.innates) bannedAbilities = bannedAbilities.concat(pokemon.innates);
+			possibleAbilities = possibleAbilities.filter(val => !bannedAbilities.includes(val));
+			if (!possibleAbilities.length) return;
+			let ability = this.getAbility(possibleAbilities[this.random(possibleAbilities.length)]);
+			this.add('-ability', pokemon, ability, '[from] ability: Power of Alchemy', '[of] ' + ally);
+			if (isAbility) {
+				pokemon.setAbility(ability);
+			} else {
+				pokemon.removeVolatile("abilitypowerofalchemy");
+				pokemon.addVolatile("ability" + ability, pokemon);
+			}
+		},
+	},
+	receiver: {
+		inherit: true,
+		onAllyFaint(ally) {
+			let pokemon = this.effectData.target;
+			if (!pokemon.hp) return;
+			let isAbility = pokemon.ability === 'receiver';
+			/**@type {string[]} */
+			let possibleAbilities = [ally.ability];
+			if (ally.innates) possibleAbilities = possibleAbilities.concat(ally.innates);
+			let bannedAbilities = ['battlebond', 'comatose', 'disguise', 'flowergift', 'forecast', 'illusion', 'imposter', 'multitype', 'powerconstruct', 'powerofalchemy', 'receiver', 'rkssystem', 'schooling', 'shieldsdown', 'stancechange', 'trace', 'wonderguard', 'zenmode'];
+			bannedAbilities.push(pokemon.ability);
+			if (pokemon.innates) bannedAbilities = bannedAbilities.concat(pokemon.innates);
+			possibleAbilities = possibleAbilities.filter(val => !bannedAbilities.includes(val));
+			if (!possibleAbilities.length) return;
+			let ability = this.getAbility(possibleAbilities[this.random(possibleAbilities.length)]);
+			this.add('-ability', pokemon, ability, '[from] ability: Receiver', '[of] ' + ally);
+			if (isAbility) {
+				pokemon.setAbility(ability);
+			} else {
+				pokemon.removeVolatile("abilityreceiver");
+				pokemon.addVolatile("ability" + ability, pokemon);
+			}
+		},
+	},
```

Comment:
This is for OM Mashups I take it?

Reply:
Yeah I was thinking about making a pullreq to allow custom game types